src/object_async/hello_async.cpp

```diff
@@ -165,8 +168,11 @@ struct AsyncHelloWorker : Nan::AsyncWorker
 {
 
     using Base = Nan::AsyncWorker;
+    // Make this class noncopyable
+    AsyncHelloWorker(AsyncHelloWorker const&) = delete;
+    AsyncHelloWorker& operator=(AsyncHelloWorker const&) = delete;
 
-    AsyncHelloWorker(bool louder, std::string const& name,
+    AsyncHelloWorker(bool louder, const std::string* name,
```

Comment:
@daniel-j-h - that is a good idea. Thanks for this back and forth - really helping me realize I need to think this through more. And improving the quality in the process!

In this case my top goal is to avoid a copy. I should spell out a bit more why: it is that I'm imagining this scenario as passing a large amount of data into the threadpool that would be a major perf hit if copied.

So I think the ideal solution here is to 1) rework node-cpp-skel to accept a `node::Buffer` since this would be the more likely the data transport from JS for large data, 2) use a `string_ref` or `string_view` approach to marshal the `node::Buffer` data into the threadpool without copying. This would allow both zero copy access and storing the `string_ref` by value on the `AsyncWorker` so that:

 - we don't need to worry about `clang-tidy` complaining about a `const & ` member
 - we don't need to arbitrarily send down a pointer to avoid the warning
 - we don't hit the -Weffc++11 warning (that happens with a pointer) like `error: ‘struct object_async::AsyncHelloWorker’ has pointer data members [-Werror=effc++] but does not override ‘object_async::AsyncHelloWorker(const object_async::AsyncHelloWorker&)’ [-Werror=effc++]`
 - we are explicit that the data is now owned by the `AsyncHelloWorker` class by using a `string_ref` - that would make it more likely to avoid programming errors with scope going forward.

I've sketched out the background and idea at https://github.com/mapbox/node-cpp-skel/issues/67